hypershift-operator/controllers/platform/aws/controller.go

```diff
@@ -56,7 +57,11 @@ type AWSEndpointServiceReconciler struct {
 
 	// controlPlaneOperatorRoleARNFn determines the control plane operator role given a hosted cluster
 	// This is used to stub the function in unit tests
-	controlPlaneOperatorRoleARNFn func(context.Context, *hyperv1.HostedCluster) (string, error)
+	controlPlaneOperatorRoleARNFn func(*hyperv1.HostedCluster) (string, error)
+
+	// controlPlaneOperatorRoleWithoutPathFn excludes the IAM path from an ARN, which is needed when adding the CPO
+	// IAM role to AWS IAM trust policies, namely the AWS VPC Endpoint Service allowed principals policy.
```

Comment:
do we really need this stub?
can we not just have a `func controlPlaneOperatorRoleARNWithoutPath()` without receiver and invoke it inline?

Reply:
I thought the stub was strange too - I'll refactor it away into a standalone function